bin/yarn

```diff
@@ -2,5 +2,9 @@
 
 set -e
 
-IMAGE_NAME=${IMAGE_NAME:-codeclimate/codeclimate-eslint}
-docker run --rm --volume "$PWD:/usr/src/app" "$IMAGE_NAME" sh -c "cd /usr/src/app && yarn $*"
+docker run \
+  --rm \
+  --user "$(id -u):$(id -g)" \
```

Comment:
I think I confirmed locally this breaks with Docker for Mac. Got this running on this branch:

```
[codeclimate-eslint]  bin/yarn upgrade                                                         devon/update-yarn-executable☃ ✱
yarn upgrade v0.21.3
error An unexpected error occurred: "EACCES: permission denied, mkdir '/.config'".
info If you think this is a bug, please open a bug report with the information provided in "/usr/src/app/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/upgrade for documentation about this command.
```

Worked as expected on `master`.

I did some conditional fanciness in one of our repos way back to set this conditionally by platform. If you think it's worth it, you could do the same here by grepping `uname -a` for `Darwin` and using root:root on Darwin, and what you already have in other cases.